common/testing/testvars/test_vars.go

```diff
@@ -39,41 +39,43 @@ import (
 type (
 	TestVars struct {
 		testName string
+		testHash uint32
+		an       Any
 		kv       sync.Map
 	}
 )
 
 func New(testName string) *TestVars {
+	th := hash(testName)
 	return &TestVars{
 		testName: testName,
+		testHash: th,
+		an:       newAny(testName, th),
 	}
 }
 
-func randString(n int) string {
-	const letterBytes = "abcdefghijklmnopqrstuvwxyz"
-	b := make([]byte, n)
-	for i := range b {
-		b[i] = letterBytes[rand.Intn(len(letterBytes))]
-	}
-	return string(b)
-}
-
 func (tv *TestVars) key(typ string, key []string) string {
 	if len(key) == 0 {
 		return typ
 	}
 	return typ + "_" + strings.Join(key, "_")
 }
 
-func (tv *TestVars) getOrCreate(typ string, key []string) string {
-	keyStr := tv.key(typ, key)
-	v, _ := tv.kv.LoadOrStore(keyStr, tv.testName+"_"+keyStr)
-	return v.(string)
+func (tv *TestVars) getOrCreate(typ string, key []string, initialVal ...any) any {
+	kvKey := tv.key(typ, key)
+	var kvVal any
+	if len(initialVal) == 0 {
+		kvVal = tv.testName + "_" + kvKey
+	} else {
+		kvVal = initialVal[0]
+	}
+	v, _ := tv.kv.LoadOrStore(kvKey, kvVal)
+	return v
```

Comment:
I can't quite put my finger on it, but I wonder if all this code would be simpler if there was a (1) string and (2) non-string version of this - most of the use cases seem to be simply strings. But it's not a big deal, this is testing code. Just wanted to mention it.